packages/devtools/lib/src/timeline/timeline_protocol.dart

```diff
@@ -361,18 +361,12 @@ class TimelineFrame {
   /// Frame start time in micros.
   int get startTime => _startTime;
   int _startTime;
-  set startTime(int t) {
-    assert(_startTime == null);
-    _startTime = t;
-  }
+  set startTime(int t) => _startTime ??= t;
 
   /// Frame end time in micros.
   int get endTime => _endTime;
   int _endTime;
-  set endTime(int t) {
-    assert(_endTime == null);
-    _endTime = t;
-  }
+  set endTime(int t) => _endTime ??= t;
```

Comment:
having a setter that quietly fails seems odd. is that intentional?

Reply:
The asserts were failing, which I found odd, because that means we can get a multiple frame start/end events with the same id - i haven't seen until now. Instead of this, I'll add an extra null check before we actually set these in the event handling code.

Reply:
sounds like the assert is catching a real bug. I would say it is worth tracking down the real cause.
An intermediate solution could be to add a comment and log the bad data to the console.
One other option is that maybe there is a logic bug and endTime is being set twice with the same value which would be less scary.

Reply:
done